### PR TITLE
feat(cbMap) AdvancedSearch BusinessMap for saving presets

### PR DESCRIPTION
**Pull request by maliknajjar**
Construct the AdvancedSearch Business Map, to define a static advanced search where we define the fields, operators and joins but not the values.

### CODE REVIEW

File: modules/Vtiger/ListView.php [hunk 1]
```diff
@@ -259,6 +259,25 @@
 $smarty->assign('Apache_Tika_URL', GlobalVariable::getVariable('Apache_Tika_URL', ''));
 $smarty->assign('ShowCreateMessage', GlobalVariable::getVariable('Application_ListView_Show_Create_Message', '0'));
 
+// send advancedSearch business map data to the frontEnd
+$advancedSearchMapResult = "MAP_NOT_FOUND";
+$map = new cbMap();
```
Comment (joebordes) [inline modules/Vtiger/ListView.php:264]: make sure this $map isn't set before and used after

File: include/js/advancefilter.js [hunk 2]
```diff
@@ -128,10 +128,16 @@
 		* (only applicable when used in filter or report context)
 		*/
 		getPreExisting: function () {
-			const input = document.getElementById('cbds-advfilt_existing-conditions'),
-				existing = JSON.parse(input.value),
-				groups = [];
-
+			let input = document.getElementById('cbds-advfilt_existing-conditions');
+			let existing = {};
+			if(input == null && advancedSearchData !== "MAP_NOT_FOUND") {
```
Comment (joebordes) [inline include/js/advancefilter.js:133]: corebos standard formatting

File: modules/cbMap/processmap/AdvancedSearch.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            </conditions>
+        </group>
+    </map>
+ *************************************************************************************************/
+
+class AdvancedSearch extends processcbMap {
+
+	public function processMap($module) {
+		$parsedxml = $this->getXMLContent();
+        $module = (string)$module[0];
```
Comment (joebordes) [inline modules/cbMap/processmap/AdvancedSearch.php:51]: formatting is incorrect here: indentation

Comment (maliknajjar) [reply modules/cbMap/processmap/AdvancedSearch.php:51]: That is strange. In my code it looks fine
image

Comment (maliknajjar) [reply modules/cbMap/processmap/AdvancedSearch.php]: I edited it from github

**Comment by joebordes** [comment]
Thanks !!

**Pull request merged**